File: GCJ/2019_round_1_b/c.py
t = int(input())
for tc in range(1, t + 1):
    n, k = map(int, input().split(' '))
    c = [int(i) for i in input().split(' ')]
    d = [int(i) for i in input().split(' ')]
    answer = 0
    for i in range(n):
        for j in range(i, n):
            c_max = d_max = 0
            for x in range(i, j + 1):
                c_max = max(c_max, c[x])
                d_max = max(d_max, d[x])
            if abs(c_max - d_max) <= k:
                answer += 1
    print('Case #{}: {}'.format(tc, answer))

File: GCJ/2019_round_1_b/c.c
#include <stdio.h>
#include <stdlib.h>

int main() {
    int t, n, k, answer, c[100005], d[100005];
    scanf("%d", &t);
    for (int test_case = 0; test_case < t; test_case++) {
        scanf("%d %d", &n, &k);
        for (int i = 0; i < n; i++) {
            scanf("%d", &c[i]);
        }
        for (int i = 0; i < n; i++) {
            scanf("%d", &d[i]);
        }
        answer = 0;
        for (int i = 0; i < n; i++) {
            for (int j = i; j < n; j++) {
                int c_max = 0, d_max = 0;
                for (int k = i; k <= j; k++) {
                    if (c_max < c[k]) {
                        c_max = c[k];
                    }
                    if (d_max < d[k]) {
                        d_max = d[k];
                    }
                }
                if (abs(d_max - c_max) <= k) {
                    answer += 1;
                }
            }
        }
        printf("Case #%d: %d\n", test_case + 1, answer);
    }
    return 0;
}
